### ISSUE TITLE
Report Missing BaseLinkedQueue Keep Member

### ISSUE THREAD

**Issue by chooblarin** [artem-zinnatullin/RxJavaProGuardRules#16]
I encountered some debug log: 

```
11-18 19:08:06.474 24887 24969 E AndroidRuntime: Caused by: java.lang.NoSuchFieldException: No field consumerNode in class Lrx/internal/util/unsafe/BaseLinkedQueueConsumerNodeRef; (declaration of 'rx.internal.util.unsafe.BaseLinkedQueueConsumerNodeRef' appears in /data/app/XXXX/base.apk)
11-18 19:08:06.474 24887 24969 E AndroidRuntime:    at java.lang.Class.getDeclaredField(Native Method)
11-18 19:08:06.474 24887 24969 E AndroidRuntime:    at rx.internal.util.unsafe.UnsafeAccess.a(UnsafeAccess.java:100)
11-18 19:08:06.474 24887 24969 E AndroidRuntime:    ... 55 more
```

I guess

```
-keepclassmembers class rx.internal.util.unsafe.BaseLinkedQueueProducerNodeRef {
   long producerNode;
   long consumerNode;
}
```

might be 

```
-keepclassmembers class rx.internal.util.unsafe.BaseLinkedQueueProducerNodeRef {
    rx.internal.util.atomic.LinkedQueueNode producerNode;
}
-keepclassmembers class rx.internal.util.unsafe.BaseLinkedQueueConsumerNodeRef {
    rx.internal.util.atomic.LinkedQueueNode consumerNode;
}
```

**Comment by artem-zinnatullin** [comment]
Thanks for report! Will investigate today.

**Comment by artem-zinnatullin** [comment]
Jesus, just remembered about this issue. I'm very sorry for the delay. Looks like you're right. Will try to launch app with these rules and check how it'll work.

**Comment by artem-zinnatullin** [comment]
@chooblarin thanks again! Released 1.0.16.1 with fix.

**Comment by lgawin** [comment]
@artem-zinnatullin, I think it's still incorrect. 
It should be:

```
-keepclassmembers class rx.internal.util.unsafe.BaseLinkedQueueProducerNodeRef {
    rx.internal.util.atomic.LinkedQueueNode producerNode;
}
-keepclassmembers class rx.internal.util.unsafe.BaseLinkedQueueConsumerNodeRef {
    rx.internal.util.atomic.LinkedQueueNode consumerNode;
}
```

as @chooblarin put in the first comment. 
Note two different classes.

**Comment by artem-zinnatullin** [comment]
oh… I have to do #4 to check all these things

@lgawin thanks, fixing…

**Comment by artem-zinnatullin** [comment]
@lgawin released 1.0.16.2, should be available in 10-30 minutes. Thanks and sorry!
